tests/pipeline/nodes/input/test_input_threading.py

```diff
@@ -54,23 +54,19 @@ def get_fps_number(avg_fps_msg: str) -> float:
     return avg_fps
 
 
-@contextmanager
-def run_pipeline_yml():
-    """Save and restore current pipeline_config.yml"""
+@pytest.fixture
+def pipeline_instance():
+    yield
     try:
-        config_saved = False
-        if os.path.isfile(PKD_PIPELINE_ORIG_PATH):
-            print("Backup existing pipeline_config.yml")
-            os.rename(src=PKD_PIPELINE_ORIG_PATH, dst=PKD_PIPELINE_BAK_PATH)
-            config_saved = True
-        yield
-    finally:
-        if config_saved:
-            print("Restore backed up pipeline_config.yml")
-            os.rename(src=PKD_PIPELINE_BAK_PATH, dst=PKD_PIPELINE_ORIG_PATH)
+        os.remove(PIPELINE_PATH)
+    except OSError as error:
+        if error.errno != errno.ENOENT:
+            # re-raise if exception isn't "no such file or directory"
+            raise
 
 
 # Unit Tests
+@pytest.mark.usefixtures("pipeline_instance")
 def test_input_threading():
     """Run input threading unit test.
```

Comment:
Using the `pipeline_instance` fixture is a much better design!
Can update the docstring below as `1.` and `.4` are no longer valid/relevant.

Reply:
Updated docstring.